packages/next/src/client/components/match-segments.ts

```diff
@@ -20,6 +21,35 @@ export const matchSegment = (
   return existingSegment[0] === segment[0] && existingSegment[1] === segment[1]
 }
 
+export const matchDynamicSegment = (
+  existingSegment: Segment,
```

Comment:
nit: is there a better for this parameter that would better indicate that the two segments' arguments are treated differently? it's not apparent from the name of the fn.

Reply:
maybe a JS doc at the top of the fn is enough